### PR TITLE
Port over G103

### PR DESCRIPTION
**Pull request by ulziibay**
Unsafe pkg is potentially dangerous. Lets warn about the usage.

### CODE REVIEW

File: go/gosec/bad_imports/unsafe.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+rules:
+  - id: avoid-using-unsafe-pkg
+    message:
+      Package `unsafe` contains operations that step around the type safety of Go programs.
+      Packages that import unsafe may be non-portable and are not protected by the Go 1 compatibility guidelines.
+    metadata:
+      cwe: 'CWE-242: Use of Inherently Dangerous Function'
+      source_rule_url: https://github.com/securego/gosec
```
Comment (minusworld) [inline go/gosec/bad_imports/unsafe.yaml:8]: Can we standardize on whether we use hyphens '-' or underscores '_'?

I think we use hyphens everywhere else. Is there a parsing reason we use underscores here?

**Comment by minusworld** [comment]
https://dev.massive.ret2.co/triager/filter/961

File: go/gosec/bad_imports/unsafe.yaml (hunk 1 above)
Comment (ulziibay) [reply go/gosec/bad_imports/unsafe.yaml:8]: Thanks for the flag. I wish we can enforce it during the rule writing process. I will address all the renaming of `source_rule_url ` in a separate PR. Sounds good?

**Comment by ulziibay** [comment]
@minusworld it is the same! I just happened to have done it and used it to pair program with a client!

File: go/gosec/bad_imports/unsafe.yaml (hunk 1 above)
Comment (minusworld) [reply go/gosec/bad_imports/unsafe.yaml:8]: Sounds good!

**Comment by ulziibay** [comment]
Closing this in favor of  #418

**Pull request closed without merging**